src/schemas/tdct/ChopperTimeStampWriter.cpp

```diff
@@ -0,0 +1,124 @@
+/** Copyright (C) 2018 European Spallation Source ERIC */
+
+/// \file
+/// \brief Implement classes required for writing chopper time stamps.
+
+#include "../../helper.h"
+
+#include "ChopperTimeStampWriter.h"
+#include "HDFFile.h"
+#include "tdct_timestamps_generated.h"
+#include <limits>
+
+namespace tdct {
+
+// Register the timestamp and name extraction class for this module
+static FileWriter::FlatbufferReaderRegistry::Registrar<ChopperTimeStampGuard>
+    RegisterSenvGuard("tdct");
+
+// Register the file writing part of this module
+static FileWriter::HDFWriterModuleRegistry::Registrar<ChopperTimeStampWriter>
+    RegisterSenvWriter("tdct");
+
+bool ChopperTimeStampGuard::verify(FlatbufferMessage const &Message) const {
+  auto Verifier = flatbuffers::Verifier(
+      reinterpret_cast<const std::uint8_t *>(Message.data()), Message.size());
+  return VerifytimestampBuffer(Verifier);
+}
+
+uint64_t
+ChopperTimeStampGuard::timestamp(FlatbufferMessage const &Message) const {
+  auto FbPointer = Gettimestamp(Message.data());
+  /// \todo This timestamp is currently EPICS epoch. This will have to be sorted
+  /// out.
```

Comment:
Create a jira ticket from \todo?

Reply:
I believe we have several JIRA tickets on this issue (I created them). I have removed the comment.